src/main/java/ru/ayaz/ChatServer.java

```diff
@@ -0,0 +1,167 @@
+package ru.ayaz;
+
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class ChatServer {
+
+    private ServerSocket serverSocket;
+    private Map<String, User> usersMap;
+    private Map<String, PrintWriter> outputWriterMap;
+
+
+    public ChatServer() {
+        try {
+            this.serverSocket = new ServerSocket(4400);
+            this.usersMap = new TreeMap<String, User>();
+            this.outputWriterMap = new TreeMap<String, PrintWriter>();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    private void sendMessageToEveryone(String message, String sender) {
+
+        System.out.println(message);
+
+        for (Map.Entry<String, User> entry : usersMap.entrySet()) {
+            String key = entry.getKey();
+            User value = entry.getValue();
+            if(!key.equals(sender)) {
+                if (!value.isInIgnoredSet(sender)) {
+                    PrintWriter pw = outputWriterMap.get(key);
+                    pw.println(message);
+                    pw.flush();
+                }
+            }
+        }
+
+    }
+
+
+    public void startChat() {
+
+        try {
+            while (true) {
+                Socket socket = serverSocket.accept();
+                Thread t = new Thread(new UserHandler(socket));
+                t.start();
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    public static void main(String[] args) {
+        ChatServer chat = new ChatServer();
+        chat.startChat();
+    }
+
+
+    public class UserHandler implements Runnable {
+
+        private User user;
+        private MessageProcessor processor;
+        private PrintWriter writer;
+        private BufferedReader reader;
+
+
+        public UserHandler(Socket socket) {
+            try {
+                InputStreamReader isReader = new InputStreamReader(socket.getInputStream());
+                this.reader = new BufferedReader(isReader);
+                this.writer = new PrintWriter(socket.getOutputStream());
+                this.user = new User();
+                this.processor = new MessageProcessor();
+
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+
+
+        public void run() {
+
+            String message;
+            String sender;
+
+            welcomeNewUser();
+
+            try {
+
+                user.setNickname(listenForMessage());
+                usersMap.put(user.getNickname(), user);
```

Comment:
By the way, updating a thread-unsafe map in a concurrent environment will lead to all sorts of strange issues, we'll address this later, just be careful to remember )